src/xrpld/app/tx/detail/AMMWithdraw.cpp

```diff
@@ -552,6 +552,29 @@ AMMWithdraw::withdraw(
         return {tecAMM_BALANCE, STAmount{}};
     }
 
+    // Check the reserve in case a trustline has to be created
+    bool const enabledFixAMMv1_2 = view.rules().enabled(fixAMMv1_2);
+    auto sufficientReserve = [&](Issue const& issue) -> TER {
+        if (!enabledFixAMMv1_2 || isXRP(issue))
+            return tesSUCCESS;
+        if (!view.exists(keylet::line(account_, issue)))
+        {
+            auto const sleAccount = view.read(keylet::account(account_));
+            if (!sleAccount)
+                return tecINTERNAL;  // LCOV_EXCL_LINE
+            auto const balance = (*sleAccount)[sfBalance].xrp();
+            auto const reserve =
+                view.fees().accountReserve((*sleAccount)[sfOwnerCount] + 1);
+
+            if (balance < reserve)
+                return tecINSUFFICIENT_RESERVE;
```

Comment:
I hate to do this after approving, but [John's comment](https://github.com/XRPLF/rippled/pull/5176#issuecomment-2455233999) raises an important point.

The reason most reserve checks use `mPriorBalance` is because that is the account's balance _before_ fees are deducted. Unless the transaction causes the account to receive XRP, that value _will_ be higher (or equal in one edge case) than the current `sfBalance` value. This is important because an account is allowed to dip into its reserves to pay fees, so we need to examine the account's balance before those fees are paid. AFAIK, this is the only transaction that can cause the transaction-sending account to receive XRP _and_ create a trustline.

This transaction is different from all(?) other transactions, though, in that it is possible for an account to receive *both* XRP *and* an IOU, which could create a trustline.

Consider a couple of scenarios (assuming account reserve is 20XRP, and owner reserve is 5XRP):
1. An account has exactly 30XRP, and a USD trustline. It submits an `AMMWithdraw` with 10 drop fee to get equal amounts of USD and EUR. Using the current rule, that transaction will fail, because the balance will be just under 30XRP. But if the account had submitted a `TrustSet` for EUR, that would have succeeded, and then the account could submit the same `AMMWIthdraw`. If we're going to allow `AMMWithdraw` to create trustlines, they should follow the same rules.
2. An account has exactly 25XRP, and no trustlines. It submits an `AMMWithdraw` with 10 drop fee to get 1000 XRP and 500 USD. Using the current rule, that transaction will succeed, because the balance will be 1025 XRP-10 drops which means we can't _just_ check the `mPriorBalance`.

Fortunately, I think the fix is simple:
```suggestion
            if (max(mPriorBalance, balance) < reserve)
                return tecINSUFFICIENT_RESERVE;
```

That also makes me realize this change neglects one other rule: The first two trust lines shouldn't count against the account's reserve, so in my example above, the transaction should succeed, even if the account only had 20XRP. I'll address that in a separate comment.